src/elexmodel/models/BootstrapElectionModel.py

```diff
@@ -1054,13 +1061,64 @@ def _is_top_level_aggregate(self, aggregate: list) -> bool:
             len(aggregate) == 2 and "postal_code" in aggregate and "district" in aggregate
         )
 
+    def _format_called_contests_dictionary(self, called_contests: dict | None) -> dict:
+        """
+        Make sure that the called contest dictionary has the correct format
+        """
+        # called_contests is a dictionary where 1 means that the LHS party has won, 0 means that the RHS party has won
+        # and -1 means that the contest is not called. If called_contests is None, assume that all contests are not called.
+        if called_contests is None or len(called_contests) == 0:
+            called_contests = {i: -1 for i in range(self.n_contests)}
+
+        if len(called_contests) != self.n_contests:
+            raise BootstrapElectionModelException(
+                f"called_contests is of length {len(called_contests)} but there are {self.n_contests} contests"
+            )
+
+        called_contest_acceptable_values = {0, 1, -1}
+        if not all(
+            any(math.isclose(value, acceptable_value) for acceptable_value in called_contest_acceptable_values)
+            in called_contest_acceptable_values
+            for value in called_contests.values()
+        ):
+            raise BootstrapElectionModelException(
+                f"called_contest values need to be either 0, 1, or -1. But current value is {called_contests}"
+            )
+
+        return called_contests
+
+    def _call_contest(self, to_call: np.array, called_contests: dict) -> np.array:
```

Comment:
Does `to_call` always contain uncalled contests, i.e. contests not yet called by the AP or by our team?  I'm trying to understand how contests that have already been called are used here.  Does our model re-call (or un-call) called contests?  Do we return all contests and their call statuses (called by model, called by others, uncalled)?  What if a user wanted to know who called the each contest? 🤔

Reply:
I'm not sure I understand your question. `to_call` is a vector of the predicted margin in all races, whether called or not. Our model does not call races at all, this function forces the vector of predictions to be on the correct side of zero when the call happens (ie. we are predicting GOP wins by 0.01% but the race is called for dems, this forces the prediction to be on the dem side).

The call status is given to the model by the user, so no need to return it.

Reply:
Ok, I think I follow.  So this method's purpose is to inform our model of the races (contests) that have been called, is that right?  I thought the idea of this method was to determine which races, based on our model's predictions/PIs, were in call status (not necessarily to call the race itself but to return such a suggestion to the user), which is why I wasn't sure what we were doing with races that had already been called 😓 

If I have any of that right now 😬 how would you feel about renaming this method to `_update_call_status()` or something similar? 🤔

Reply:
Specifically updating the point prediction (the prediction intervals get updated in the function specific to the aggregate prediction intervals). I don't love that name, call status is not really a phrase we've used for the model before. How about `adjust_predictions`?

Reply:
Sure! 🎉   Or maybe `_adjust_predictions_based_on_race_calls()`?  Hmm that's a lot of letters.  `_adjust_called_contests()`? 🤔

Reply:
done!